### DISCUSSION TITLE
CrystalConv output channels

### DISCUSSION THREAD

**Discussion by YuraSumy** [danielegrattarola/spektral#316] [Q&A] [answered]
**CrystalConv** layer works only if the number of input channels is equal to the number of output channels.
self.conv1 = CrystalConv(2, activation="elu", kernel_regularizer=l2(l2_reg))
2 is the dimension of input node features, if to change to 64, the exception occurs:

 ValueError: Dimensions must be equal, but are 2 and 64 for '{{node graph_model/crystal_conv/add}} = AddV2[T=DT_FLOAT](graph_model/Cast, graph_model/crystal_conv/transpose_1)' with input shapes: [32,15,2], [32,15,64].

32 - batch size, 15 - nodes number, 2 - input node channels, 64 is the number of output channels

Is it expected behavior?

**Comment by danielegrattarola** [comment, marked as answer]
Hi, 

yes, that is by design. There shouldn't even be an option to control the number of output channels, it should be automatically determined by the layer. I will change it in the next update, thanks for pointing this out to me.
